### DISCUSSION TITLE
I need help understanding the codebase

### DISCUSSION THREAD

**Discussion by HayataSama** [monkeytypegame/monkeytype#1528] [Questions] [answered]
Hi! 
so I am currently trying to learn javascript and I've been using monkeytype for more than a year now and I love it!
I thought it's a good opportunity to contribute to monkeytype and learn some new stuff along the way by reading other people's code.
my problem is that it's my first time reading a big project and I have no clue where to start or how to understand the codebase.
I would say I'm good at HTML and CSS but I don't know javascript that well

**Comment by Miodec** [comment, marked as answer]
You could start by getting the local development setup by following the instructions in `contributing.md`. But I think the code is relatively readable (compared to couple months ago). Most things are in small modules which are named appropriately.

Also, more specific questions would help me help you. Feel free to ask.

**Reply by HayataSama** [reply]
thank you so much for your reply!
honestly, I don't even know with which file I should start. there is just so much stuff that is going on and for me (as a beginner) is kind of overwhelming.
